### ISSUE TITLE
about “image2depth_semantickitti” in homemade dataset

### ISSUE THREAD

**Issue by wjq777777** [vita-epfl/VoxDet#5]
Hello！ This is a great job. We are following your work. Similarly, we also encountered it when reproducing CGFormer. We organized our own dataset in the format of the semantickitti dataset, but encountered difficulties when processing the data using docs/dataset.md. Among them, "image2depth_semantickitti to create depth maps". Since the images in our dataset are different from those in semantickitti, is it necessary to pre-train a depth estimation model to perform the above step? If so, how do I need to train a depth estimation model? Could you please provide relevant tutorials?

Looking forward to your reply. May everything go well!

**Comment by wjq777777** [comment]
Thank you for your prompt reply. As you said, the depth preprocessing is referenced from Voxformer, which uses a stereo metric depth estimator pre-trained on the KITTI dataset. The dataset I made myself is quite different from the kitti dataset. So, should I pre-train a depth estimation model on my dataset first and then conduct Depth pre-processing? Or can I directly use the steps of Voxformer to generate depth data?

Everything goes well!

**Comment by wymanCV** [comment]
Hi, if you have depth ground-truth, it will be great to fine-tune the depth estimator with your custom data. Otherwise, you can directly use existing metric depth foundational models to obtain the depths, which seems to be a much easier way.

**Comment by wjq777777** [comment]
Thank you for your reply! Unfortunately, I don't have the data of depth ground-truth. According to what you said, I can directly use the existing basic model for measuring depth to obtain depth. But one point I hesitate about is: My image data is an unstructured scene. Will there be a significant gap if I directly use the existing basic model?

It would be wonderful if this were feasible!

Looking forward to your reply. Wish you all the best!

**Comment by wymanCV** [comment]
Hi, I think that the foundational depth model (mono or stereo), e.g., Metric3D, should have a better generalization ability than dataset-sepecific ones. Considering that you don’t have depth to fine-tune a depth estimator, using foundaitional might be one of solutions. Currently, I don't have beter solutions. Maybe, I can help you @pkqbajng (CGFormer), who better knows 3D vision than me. Thank you!
